src/packages/media/media/property-editors/media-picker/property-editor-ui-media-picker.element.ts

```diff
@@ -30,21 +31,41 @@ export class UmbPropertyEditorUIMediaPickerElement extends UmbLitElement impleme
 		return undefined;
 	}
 
+	@state()
+	_items: Array<string> = [];
+
 	@state()
 	private _limitMin: number = 0;
 	@state()
 	private _limitMax: number = Infinity;
 
+	protected firstUpdated(_changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
+		super.firstUpdated(_changedProperties);
+
+		this._items = this.value ? this.value.map((x) => x.mediaKey) : [];
```

Comment:
Consider moving this to a _setter_ instead (`public set value()`) which will ensure it happens every time the property is updated and not just when the component itself is updated.